docs/asciidoc/modules/ROOT/pages/ml/watsonai.adoc

```diff
@@ -0,0 +1,145 @@
+[[watson-api]]
+= watson API Access
+:description: This section describes procedures that can be used to access the watson API.
+
+NOTE: You need to acquire an https://cloud.ibm.com/apidocs/watson-data-api#creating-an-iam-bearer-token[Watson bearer access token] to use these procedures.
+This token must be defined in the second parameter
+
+
+Each procedure can have the following APOC config, i.e. in `apoc.conf` or via docker env variable
+
+.Apoc configuration
+|===
+|key | description | default
+| apoc.ml.watson.project.id | The project_id | the `project_id` defined in the configuration map
+| apoc.ml.watson.url | The REST API endpoint | https://eu-de.ml.cloud.ibm.com/ml/v1-beta/generation/text?version=2023-05-29
+|===
+
+We can put the body request directly into the configuration parameter, 
+except for the `"input"` key, which will be added through the first parameter of the procedure, as we will see later.
+
+For example, if we want to send a request this request:
+```
+{
+ "model_id": "google/flan-ul2",
+ "input": "my test input",
+ "parameters": {
+  "max_new_tokens": 3
+ },
+ "project_id": "MyProjectId"
+}
+```
+we can create the following configuration map:
+```
+{
+    model_id: "google/flan-ul2",
+    project_id: "MyProjectId",
+    parameters: {
+      max_new_tokens: 3
+    }
+}
+```
+
+Note that the `model_id` is not mandatory, since has a default value `"ibm/granite-13b-chat-v2"`
+
+Also the `project_id` is not mandatory, if it has been defined via APOC configuration,
+or if alternatively is present in the configuration map an entry with the key `space_id` or `wml_instance_crn`; otherwise, an error will be thrown, since the Watson Request APIs require one of these.
+
+In addition, we can put in the configuration map, the entry `endpoint: "ENDPOINT_URL`, 
+which define the REST API endpoint, and takes precedence over the `apoc.watson.url` APOC configuration.
+If neither is specified, the default value `"https://eu-de.ml.cloud.ibm.com/ml/v1-beta/generation/text?version=2023-05-29"` will be used.
+
+
+The following examples assume we have this `apoc.conf`:
+```
+apoc.watson.project.id=MY_PROJECT_ID
+```
+
+== Text Completion API
```

Comment:
Perhaps add a section on Embedding APIs that states that Watson doesn't have any built-in embedding APIs and that one use another service for generating embeddings, like one of the other ML integrations or HuggingFace embeddings like sentence-transformers.